main.tf

```diff
@@ -1,65 +1,69 @@
-module "ipam" {
-  source  = "./modules/ipam"
-  tags    = var.ipam.tags
-  cascade = var.ipam.cascade
+resource "aws_vpc_ipam" "this" {
+  description = var.vpc_ipam.ipam.description
+  operating_regions {
+    region_name = var.vpc_ipam.ipam.region_name
+  }
```

Comment:
this `operating_regions` is likely plural, so need to use a list of objects as the object type, and use dynamic to expand out the blocks
```
  dynamic "operating_regions" {
    for_each = var.vpc_ipam.ipam.operating_regions
    content {
      region_name = each.value.region_name
    }
  }
```